tests/lax_numpy_ufuncs_test.py

```diff
@@ -226,13 +229,18 @@ def test_binary_ufunc_outer(self, name, lhs_shape, rhs_shape, dtype):
   @jtu.sample_product(
       SCALAR_FUNCS,
       [{'shape': shape, 'axis': axis}
-       for shape in nonscalar_shapes
+       for shape in all_shapes
        for axis in [None, *range(-len(shape), len(shape))]],
       dtype=jtu.dtypes.floating,
   )
   def test_frompyfunc_reduce(self, func, nin, nout, identity, shape, axis, dtype):
     if (nin, nout) != (2, 1):
       self.skipTest(f"reduce requires (nin, nout)=(2, 1); got {(nin, nout)=}")
+    if identity is None and axis is None and 0 in shape:
+      self.skipTest("reduce along zero size array for ufuncs with no identity.")
+    if identity is None and axis is not None and shape[axis] == 0:
+      self.skipTest("reduce along zero-sized axis for ufuncs with no identity.")
```

Comment:
Rather than skipping these tests, we should use `assertRaises` to assert that they raise the expected error